pkgs/development/libraries/paho-mqtt-c/default.nix

```
@@ -21,7 +21,11 @@ stdenv.mkDerivation rec {
 
   buildInputs = [ openssl ];
 
-  cmakeFlags = [ "-DPAHO_WITH_SSL=TRUE" ];
+  cmakeFlags = [
+    (lib.strings.cmakeBool "PAHO_WITH_SSL" true)
+    (lib.strings.cmakeBool "PAHO_BUILD_STATIC" enableStatic)
+    (lib.strings.cmakeBool "PAHO_BUILD_SHARED" enableShared)
```

```suggestion
    (lib.cmakeBool "PAHO_WITH_SSL" true)
    (lib.cmakeBool "PAHO_BUILD_STATIC" enableStatic)
    (lib.cmakeBool "PAHO_BUILD_SHARED" enableShared)
```
See https://github.com/NixOS/nixpkgs/blob/f35896eab21738816b9e374703209d74ca215836/lib/default.nix#L109